test/runtime_tests.cc

```diff
@@ -1898,5 +1898,62 @@ namespace detail {
+
+			    accessor acc2{buf_a, cgh, one_to_one<1>(), cl::sycl::read_only};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::read, cl::sycl::target::device>, decltype(acc2)>);
+
+			    accessor acc3{buf_a, cgh, one_to_one<1>(), cl::sycl::read_write};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::read_write, cl::sycl::target::device>, decltype(acc3)>);
+
+			    accessor acc4{buf_a, cgh, one_to_one<1>(), cl::sycl::write_only, cl::sycl::no_init};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::discard_write, cl::sycl::target::device>, decltype(acc4)>);
+
+			    accessor acc5{buf_a, cgh, one_to_one<1>(), cl::sycl::read_write, cl::sycl::no_init};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::discard_read_write, cl::sycl::target::device>, decltype(acc5)>);
+		    },
+		    buf_a.get_range());
+
+		const auto tsk = tm.get_task(tid);
+		REQUIRE(tsk->get_buffer_access_map().get_access_modes(0).count(cl::sycl::access_mode::write) == 1);
+		REQUIRE(tsk->get_buffer_access_map().get_access_modes(0).count(cl::sycl::access_mode::read) == 1);
+		REQUIRE(tsk->get_buffer_access_map().get_access_modes(0).count(cl::sycl::access_mode::read_write) == 1);
+		REQUIRE(tsk->get_buffer_access_map().get_access_modes(0).count(cl::sycl::access_mode::discard_write) == 1);
+		REQUIRE(tsk->get_buffer_access_map().get_access_modes(0).count(cl::sycl::access_mode::discard_read_write) == 1);
+	}
+
+	TEST_CASE("accessors deduces correctly from SYCL 2020 tags for host", "[accessor]") {
+		distr_queue q;
+		std::vector mem_a{42};
+		buffer<int, 1> buf_a(mem_a.data(), cl::sycl::range<1>{1});
+		q.submit([=](handler& cgh) {
+			accessor a{buf_a, cgh, fixed<1>({0, 1}), cl::sycl::read_write_host_task};
+			accessor acc1{buf_a, cgh, one_to_one<1>(), cl::sycl::write_only_host_task};
+			static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::write, cl::sycl::target::host_buffer>, decltype(acc1)>);
+
+			accessor acc2{buf_a, cgh, one_to_one<1>(), cl::sycl::read_only_host_task};
+			static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::read, cl::sycl::target::host_buffer>, decltype(acc2)>);
+
+			accessor acc3{buf_a, cgh, one_to_one<1>(), cl::sycl::read_write_host_task};
+			static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::read_write, cl::sycl::target::host_buffer>, decltype(acc3)>);
+
+			accessor acc4{buf_a, cgh, one_to_one<1>(), cl::sycl::write_only_host_task, cl::sycl::no_init};
+			static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::discard_write, cl::sycl::target::host_buffer>, decltype(acc4)>);
+
+			accessor acc5{buf_a, cgh, one_to_one<1>(), cl::sycl::read_write_host_task, cl::sycl::no_init};
+			static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::discard_read_write, cl::sycl::target::host_buffer>, decltype(acc5)>);
+
+			cgh.host_task(on_master_node, [=] { ++a[{0}]; });
+		});
```

Comment:
Why is `task::get_buffer_access_map` not checked here? Actually I think you could probably share that part with the device accessors, by doing a single `TEST_CASE` for both, and then have a `SECTION("for device accessors")` and `SECTION("for host task accessors")`. See for example the test "buffer_manager remember coherence replications between consecutive accesess".

Reply:
done.